grammar/haxe.bnf

```diff
@@ -375,10 +377,14 @@ varInit ::= '=' expression {pin=1}
 parameter ::= '?'? componentName typeTag? varInit?
 {mixin="com.intellij.plugins.haxe.lang.psi.impl.HaxeParameterPsiMixinImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeParameterPsiMixin"}
 
-parameterList ::= parameter? (',' parameter)* {recoverWhile="parameterListRecovery" mixin="com.intellij.plugins.haxe.lang.psi.impl.HaxeParameterListPsiMixinImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeParameterListPsiMixin"}
-private parameterListRecovery ::= !')' // after parameterList always ')'
+openParameterList ::= parameter {extends=parameterList}
+private parenthesizedParameterList ::= '(' parameterList ')' //{pin=1} // pin=1 to force matching the parens.
+parameterList ::= parameter? (',' parameter)* {mixin="com.intellij.plugins.haxe.lang.psi.impl.HaxeParameterListPsiMixinImpl" implements="com.intellij.plugins.haxe.lang.psi.HaxeParameterListPsiMixin"}
+
+private arrowFunctionLiteral ::= arrowFunctionDeclaration
+private keywordFunctionLiteral ::= 'function' parenthesizedParameterList typeTag? 'untyped'? functionCommonBody {pin=2}
```

Comment:
Not really related to this change, but I've just noticed it: That `'untyped'?` does not belong here. Untyped is part of the general expression syntax, so you should have something like:

```
function ::= 'function' args typeTag? expr
expr ::= 'untyped' expr
expr ::= ...
```